htmd/qm/base.py

```diff
@@ -0,0 +1,202 @@
+
+        self._arg('molecule', ':class: `htmd.parameterization.ffmolecule.FFMolecule`', 'Molecule',
+                  validator=val.Object(FFMolecule), required=True)
+        self._arg('multiplicity', 'int', 'Multiplicity of the molecule',
+                  default=1, validator=val.Number(int, 'POS'))
+        self._arg('theory', 'str', 'Level of theory',
+                  default='B3LYP', validator=val.String(), valid_values=self.THEORIES)
+        self._arg('correction', 'str', 'Empirical dispersion correction',
+                  default='none', validator=val.String(), valid_values=self.CORRECTIONS)
+        self._arg('basis', 'str', 'Basis set',
+                  default='6-31G*', validator=val.String(), valid_values=self.BASIS_SETS)
+        self._arg('solvent', 'str', 'Implicit solvent',
+                  default='vacuum', validator=val.String(), valid_values=self.SOLVENTS)
+        self._arg('esp_points', ':class: `numpy.ndarray`', 'Point to calculate ESP',
+                  default=None, nargs='*')  # TODO implement validator
+        self._arg('optimize', 'boolean', 'Optimize geometry',
+                  default=False, validator=val.Boolean())
+        self._arg('restrained_dihedrals', ':class: `numpy.ndarray`',
+                  'List of restrained dihedrals (0-based indices)',
+                  default=None, nargs='*')  # TODO implement validator
+        self._arg('queue', ':class:`SimQueue <htmd.queues.simqueue.SimQueue>` object',
+                  'Queue object used to run simulations',
+                  default=LocalCPUQueue())
+        self._arg('directory', 'str', 'Working directory',
+                  default='.', validator=val.String())
+
+    @property
+    @abstractmethod
+    def _command(self):
+        pass
+
+    @abstractmethod
+    def _completed(self, directory):
+        pass
+
+    @abstractmethod
+    def _writeInput(self, directory, iframe):
+        pass
+
+    @abstractmethod
+    def _readOutput(self, directory):
+        pass
+
+    def _setup(self):
+
+        # Set up the molecule
+        # TODO remove molecule coping!
+        self._molecule = self.molecule.copy()
+        self._nframes = self._molecule.coords.shape[2]
+        self._natoms = self._molecule.coords.shape[0]
+        self._charge = self._molecule.netcharge
+
+        # Set up ESP points
+        if self.esp_points is not None:
+            # TODO move to a validator
+            if self.esp_points.shape[1] != 3:
+                raise ValueError("ESP point array must be npoints x 3")
+            if self._molecule.coords.shape[2] != 1:
+                raise ValueError("Can only specift ESP point array with a single frame of coords")
+
+        # Set up restrained dihedrals
+        self._restrained_dihedrals = None
+        if self.restrained_dihedrals is not None:
+            self._restrained_dihedrals = self.restrained_dihedrals + 1  # Convert to 1-based indices
+
+        # TODO extract from SimQueue object
```

Comment:
This can be done now

Reply:
and I mean, replace directly at the necessary places `self._ncpus` for `self.queue.ncpus`

Reply:
Pending...